### PR TITLE
Ensure generated date facet links are HTML-safe

### LINKED ISSUE #393: In metadata display, unexpected characters are showing up in Date field
See screenshot 
Screen Shot 2022-12-09 at 4 26 32 PM

### PR DESCRIPTION
**Pull request by thatbudakguy**
Fixes #393

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -38,11 +38,12 @@ def component_media_type_label(value:, **_kwargs)
...
   def render_date_facet_links(value:, pattern: /(\d{4}(?:-\d{2}-\d{2})?)/, sep: ', ', **_kwargs)
-    value.map do |date|
-      parts = date.split(pattern).map do |part|
+    html = value.map do |date|
+      date.split(pattern).map do |part|
         part.match(pattern) ? link_to(part, search_action_path(search_state.filter('date').add(part))) : part
-      end
-      safe_join(parts)
+      end.join
     end.join(sep)
```
Comment (jcoyne) [inline app/helpers/application_helper.rb:45]: I think the problem was that you used a join here and not safe_join.

Comment (jcoyne) [reply app/helpers/application_helper.rb:45]: You probably also want to add a test for `html_safe?`

Comment (thatbudakguy) [reply app/helpers/application_helper.rb:45]: oh, they both have to be safe_join? I thought joining safe parts would be safe, but I guess not!

is there a matcher that lets me assert that something is html safe?

Comment (jcoyne) [reply app/helpers/application_helper.rb:45]: yes, you need to do `expect(value).to be_html_safe` or `expect(value.html_safe?).to be true`

Comment (thatbudakguy) [reply app/helpers/application_helper.rb:45]: perfect, thx

**Review by jcoyne — APPROVED** [review]

**Pull request merged**